pkgs/applications/networking/instant-messengers/deltachat-desktop/default.nix

```diff
@@ -0,0 +1,132 @@
+{ lib
+, coreutils
+, electron
+, esbuild
+, fetchFromGitHub
+, fetchgit
+, fetchpatch
+, fetchurl
+, makeDesktopItem
+, makeWrapper
+, nix-gitignore
+, nodejs
+, nodePackages
+, pkgs
+, python2
+, runCommand
+, stdenv
+, writeTextFile
+}:
+
+let
+  nodeEnv = import ../../../../development/node-packages/node-env.nix {
+    inherit lib stdenv nodejs python2 pkgs runCommand writeTextFile;
+    libtool = if stdenv.isDarwin then pkgs.darwin.cctools else null;
+  };
+
+  # obtained using `node2nix --development`
+  # `--lock` does not work due to https://github.com/svanderburg/node2nix/issues/247
+  deltachatNodePackages = import ./node-packages.nix {
```

Comment:
This is 12k lines quite huge for a single package.

Reply:
Without development mode, it would be just 1746. Do you think it makes sense to try specifying only what's needed as supplement inputs? See https://github.com/svanderburg/node2nix/#adding-additionalglobal-npm-packages-to-a-packaging-process.

Reply:
I guess I should use the NUR until [nodelock2nix](https://github.com/tweag/npmlock2nix) is inlcuded in Nixpkgs.

Reply:
@Mic92 If my upstream PR https://github.com/deltachat/deltachat-desktop/pull/2314 is accepted, this can be reduced to about 5k. Do you think that's more acceptable?

Reply:
Also see regarding npmlock2nix: https://github.com/NixOS/nixpkgs/pull/128749#issuecomment-888108492